### ISSUE TITLE
Multiple files

### ISSUE THREAD

**Issue by Aizenko** [balderdashy/skipper#45]
Hi,
Uploading one file at a time is working. But if I want to put several files in the same upload it crashed with

`verbose: Error: EMAXBUFFER: An Upstream ('file0') timed out before it was plugged into a receiver. It was still unused after waiting 2500ms. You can configure this timeout by changing the 'maxTimeToBuffer' option.`
`verbose: Error: EMAXBUFFER: An Upstream ('file1') timed out before it was plugged into a receiver. It was still unused after waiting 2500ms. You can configure this timeout by changing the 'maxTimeToBuffer' option.`

I try same name for the input 'file' or file[],  file[0] and file[1] and file0 end file1... nothing is working.

Regards,

**Comment by mikermcneil** [comment]
@Alzenko would you post the code you're using?  I'd do it like:

```js
someAction: function (req, res) {


var uploadedFilesOfType0;
var uploadedFilesOfType1;

  async.parallel([
    function (next) {
      req.file('fileType0').upload(function (err, _uploadedFilesOfType0){
        if (err) return next(err);
        uploadedFilesOfType0 = _uploadedFilesOfType0;
        return next();
      });
    },
    function (next) {
      req.file('fileType1').upload(function (err, _uploadedFilesOfType1) {
        if (err) return next(err);
        uploadedFilesOfType1 = _uploadedFilesOfType1;
        return next();
      });
    }
  ], function afterwards(err) {
    if (err) return res.negotiate(err);

    // `uploadedFilesOfType1` and `uploadedFilesOfType0` contain the uploaded metadata

    return res.ok();
  });
}
```

**Comment by mikermcneil** [comment]
My guess is that the reason you're getting EMAXBUFFER is because if you put the second `req.file(..).upload()` inside of the callback for the first, Skipper is having to hold on to incoming files from the other field if they come in first.  There's actually not a whole lot wrong with this since streams2 in node implement backpressure (it's not like you're buffering TBs of files in memory-- node pushes back on the upload to keep more bytes from flowing in until it's ready), but EMAXBUFFER is there to protect you as a failsafe.  If the serial behavior is actually what you want to enforce (i.e. for business reasons), you can increase this timeout as an option

**Comment by penciltim** [comment]
@mikermcneil I've encounter this problem, and your guess is right. I put one upload inside another.
Your solution using async is perfect, thanks!!!

**Comment by sailsbot** [comment]
Thanks for posting, @Aizenko.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
